File: dev-docs/snippets/workflows/get-pending-applies.swift
import JsBaoClient

// Fetch runs awaiting client-side apply for a document. Useful on reconnect
// to recover applies whose events were missed while offline.
func getPendingApplies(client: JsBaoClient, contextDocId: String) async throws {
  // #region example
  let pending = try await client.workflows.getPendingApplies(contextDocId: contextDocId)
  // pending: [PendingApplyInfo] — one entry per run still in apply_pending
  // (read pending[i].runKey / .workflowKey / .contextDocId / .meta)
  // #endregion example
  _ = pending
}

File: dev-docs/snippets/workflows/get-pending-applies.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Fetch runs awaiting client-side apply for a document. Useful on reconnect
// to recover applies whose events were missed while offline.
export async function getPendingApplies(client: JsBaoClient, contextDocId: string) {
  // #region example
  const pending = await client.workflows.getPendingApplies({ contextDocId });
  // pending: one entry per run still in apply_pending
  // #endregion example
  return pending;
}
